src/plugin/custom/summernote-custom-font.js

```diff
@@ -0,0 +1,142 @@
+                const styleInfo = context.invoke('editor.currentStyle');
+                if (options.addDefaultFonts) {
+                    $.each(styleInfo['font-family'].split(','), (idx, fontname) => {
+                        fontname = fontname.trim().replace(/['"]+/g, '');  
+                        if (context.invoke('buttons.isFontDeservedToAdd',fontname)) {
+                            if (fontNames.indexOf(fontname) === -1) {
+                                fontNames.push(fontname);
+                            }
+                        }
+                    });
+                }
+
+                $customFontBtn = ui.buttonGroup([
+                    ui.button({
+                        className: 'dropdown-toggle',
+                        contents: ui.dropdownButtonContents(
+                            '<span class="note-current-fontname"></span>', options
+                        ),
+                        tooltip: lang.customFont.name,
+                        data: {
+                            toggle: 'dropdown',
+                        },
+                    }),
+                    ui.dropdownCheck({
+                        className: 'dropdown-fontname',
+                        checkClassName: options.icons.menuCheck,
+                        items: fontNames,
+                        title: lang.customFont.name,
+                        template: (item) => {
+                            if (typeof item === 'object') {
+                                return '<span style="font-family: ' + item.value + '">' + item.name + '</span>';
+                            } else if (typeof item === 'string') {
+                                return '<span style="font-family: ' + item + '">' + item + '</span>';
+                            }
+                        },
+                        click: function (event) {
+                            context.createInvokeHandler('customFont.setFontName')(event)
+                            context.invoke('customFont.updateCurrentStyle')
+                        }
+                    }),
+                ]).render();
+                return $customFontBtn
+            })
+
+            this.setFontName = function (value) {
+                context.invoke("beforeCommand");
+                modules.editor.fontStyling.call(modules.editor, 'font-family', value)
+                context.invoke("afterCommand");
+            }
+
+            this.updateCurrentStyle = function(){
+                const styleInfo = context.invoke('editor.currentStyle');
+                if (styleInfo['font-family']) {
+                    const fontNames = styleInfo['font-family'].split(',').map((name) => {
+                        return name.replace(/[\'\"]/g, '')
+                        .replace(/\s+$/, '')
+                        .replace(/^\s+/, '');
+                    });
+                    const showFontName = Array.isArray(fontNames) ? fontNames[0] : fontNames
```

Comment:
目前寫法無法對應到 system-ui，以 `buttons.isFontInstalled` 對應已安裝的字體，紀錄該字體到 `updateFontName` 待之後更新

```suggestion
                    const showFontName = lists.find(fontNames, modules.buttons.isFontInstalled.bind(modules.buttons))
                    var updateFontName = { name: showFontName, value: showFontName }
```

Reply:
Fixed at https://github.com/iq-service-inc/react-summernote/pull/3/commits/476fc07e4bd56ae330ba5c21c1364510d161cfeb